### ISSUE TITLE
Code Contribution: 【Lv3】【Operator Development】batch_norm_backward

### ISSUE THREAD

**Issue by StrongSpoon** [FlagOpen/FlagGems#315]
## Description 任务介绍
Develop backward function for batch_norm operator.
开发batch_norm算子的反向功能。
## Requirements 任务要求
__Interface 接口__
`batch_norm_backward(Tensor grad_out, Tensor input, Tensor weight, Tensor? running_mean, Tensor? running_var, Tensor? save_mean, Tensor? save_var, bool update, float eps, bool[3] output_mask, Tensor reserve) -> (Tensor, Tensor, Tensor)`
__Function reference 功能参考__
https://pytorch.org/docs/stable/generated/torch.nn.functional.batch_norm.html
__Implementation reference 实现参考__
https://github.com/FlagOpen/FlagGems/blob/master/src/flag_gems/ops/groupnorm.py
https://github.com/FlagOpen/FlagGems/blob/master/src/flag_gems/ops/layernorm.py

The operator should support all optional arguments defined in the interface and return the corresponding number of values. 
算子应支持接口中定义的所有参数选项，并返回相应数量的值。
## DDL 提交时间
Please submit a Pull Request within 3 weeks after accepting the assignment.
请于接取任务后三周内提交PR。
Please provide both accuracy test and performance test code.
请同时提供实现正确性测试与性能测试代码。

**Comment by 2niuhe** [comment]
2niuhe认领
